src/js/generator-controller.js

```diff
@@ -0,0 +1,222 @@
+        artifactId: "demo",
+        fractions : function(fractions) {
+          return $scope.fractions.filter(function(item){return item.selected;});
+        }
+    }
+
+    configureSearchEngine(allFractions);
+
+    $scope.generate = function(model) {
+       saveAs(createZip(model), "swarm-example.zip");
+    }
+
+    $scope.viewDeps = false;
+    $scope.toggleViewDeps = function(val) {
+      $scope.viewDeps = val;
+    }
+});
+
+extractCategories = function(fractions) {
+  var categories = [];
+  fractions.forEach(function (fraction) {
+    fraction.category = fraction.category || fraction.tags.split(",")[0] || "General";
+    if (fraction.category && categories.indexOf(fraction.category) == -1) {
+      categories.push(fraction.category);  
+    }
+  });
+  return categories.sort();
+}
+
+configureSearchEngine = function(fractions) {
+  var searchEngine = new Bloodhound({
+    datumTokenizer: Bloodhound.tokenizers.obj.nonword('name', 'description', 'tags'),
+    queryTokenizer: Bloodhound.tokenizers.whitespace,
+    identify: function (obj) {
+        return obj.name;
+    },
+    cache: false
+  });
+  
+  updateSearchEngine(searchEngine, fractions);
+
+  $('.typeahead').typeahead({
+      minLength: 2,
+      highlight: true
+    },{
+      display: 'name',
+      source: searchEngine,
+      templates: {
+        suggestion: function (data) {
+          return "<div><strong>" + data.name + "</strong><br/><small>" + data.description + "</small></div>";
+         }
+      }
+  });
+
+  $('#search').bind('typeahead:select', function (ev, suggestion) {
+      suggestion.selected = true;
+      $('#search').typeahead('val', '');
+      updateSearchEngine(searchEngine, fractions.filter(function(item){return !item.selected;}));
+      // Update AngularJS model
+      angular.element($('#search')).scope().$apply();
+  });  
+}
+
+updateSearchEngine = function (searchEngine, fractions) {
+  searchEngine.clear();
+  searchEngine.add(fractions);
+} 
+
+createZip = function(model) {
+  var pom = createPOM(model);
+  var zip = new JSZip();
+  var projectRoot = zip.folder(model.artifactId || "demo");
+  projectRoot.file("pom.xml", pom);
+  var srcJavaFolder = projectRoot.folder("src").folder("main").folder("java");
+  if (isJAXRSSelected(model)) {    
+    var restFolder = srcJavaFolder.folder("com").folder("example").folder("rest");
+    restFolder.file("HelloWorldEndpoint.java", createRestEndpoint());
+    restFolder.file("RestApplication.java", createRestApplication());
+  }
+  return zip.generate({type:"blob"});
+}
+
+isJAXRSSelected = function(model) {
+  var selectedFractions = model.fractions();
+  var result = (selectedFractions.length == 0); 
+  for (i=0;i<selectedFractions.length;i++) {
+    if (selectedFractions[i].name === 'JAX-RS') {
+      result = true;
+    } 
+  }
+  return result;
+} 
+
+createPOM = function(model) {
+  var groupId = model.groupId || "com.example";
+  var artifactId = model.artifactId || "demo";
+  var pom = ''
```

Comment:
Do you think this can become an angular template, i.e. external to the js? That would simplify maintenance ...